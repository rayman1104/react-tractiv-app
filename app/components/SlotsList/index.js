/**
 *
 * SlotsList
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import SurfImage from 'images/1_surf.png';
import SurfImage2x from 'images/1_surf@2x.png';
import HikeImage from 'images/2_hike.png';
import HikeImage2x from 'images/2_hike@2x.png';
import WeightsImage from 'images/3_weights.png';
import WeightsImage2x from 'images/3_weights@2x.png';
import SpinImage from 'images/4_spin.png';
import SpinImage2x from 'images/4_spin@2x.png';

const activityImages = {
  SURFING: [SurfImage, SurfImage2x],
  HIKING: [HikeImage, HikeImage2x],
  WEIGHTS: [WeightsImage, WeightsImage2x],
  SPINNING: [SpinImage, SpinImage2x],
};

function byField(field) {
  return (a, b) => (a[field] > b[field] ? 1 : -1);
}

/* eslint-disable react/prefer-stateless-function */
class SlotsList extends React.PureComponent {
  render() {
    const { slots } = this.props;
    const dates = Object.keys(slots).sort();
    if (dates.length < 1) {
      return (
        <span className="app_main_section_text">
          You don&apos;t have any activities schedules yet.
        </span>
      );
    }
    return dates.map(date => (
      <section key={date} className="app_main_day">
        <h2 className="app_main_day_date">{moment(date).format('MMMM D')}</h2>
        <h2 className="app_main_day_weekday">{moment(date).format('dddd')}</h2>
        <div className="app_main_day_roll_wrap">
          <div className="app_main_day_roll">
            {slots[date].sort(byField('date')).map(slot => (
              <div
                key={slot.date + slot.activity}
                className="app_main_day_activity"
              >
                <div className="app_grid_day_action">
                  <img
                    src={activityImages[slot.activity][0]}
                    srcSet={`${activityImages[slot.activity][1]} 2x`}
                    className="app_grid_day_action_icon"
                    alt={slot.activity}
                  />
                </div>
                <span className="app_grid_day_time">{slot.time}</span>
              </div>
            ))}
          </div>
        </div>
      </section>
    ));
  }
}

SlotsList.propTypes = {
  slots: PropTypes.objectOf(
    PropTypes.arrayOf(
      PropTypes.shape({
        activity: PropTypes.oneOf(['SURFING', 'HIKING', 'WEIGHTS', 'SPINNING'])
          .required,
        time: PropTypes.string.required,
        date: PropTypes.string.required,
        duration: PropTypes.number,
      }),
    ),
  ),
};

export default SlotsList;
