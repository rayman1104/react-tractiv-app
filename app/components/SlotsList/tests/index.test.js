import React from 'react';
import { shallow } from 'enzyme/build';
import { shallowToJson } from 'enzyme-to-json';

import SlotsList from '../index';

describe('<SlotsList />', () => {
  it('Should render correctly', () => {
    const mockSlots = [
      {
        activity: 'SURFING',
        date: '2019-03-29T11:00',
        duration: 90,
      },
      {
        activity: 'WEIGHTS',
        date: '2019-03-27T13:00',
        duration: 90,
      },
    ];
    const output = shallow(<SlotsList slots={mockSlots} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
