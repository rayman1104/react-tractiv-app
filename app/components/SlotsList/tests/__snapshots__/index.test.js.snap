// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlotsList /> Should render correctly 1`] = `
Array [
  <section
    className="app_main_day"
    key="2019-03-27"
  >
    <h2
      className="app_main_day_date"
    >
      March 27
    </h2>
    <h2
      className="app_main_day_weekday"
    >
      Wednesday
    </h2>
    <div
      className="app_main_day_roll_wrap"
    >
      <div
        className="app_main_day_roll"
      >
        <div
          className="app_main_day_activity"
          key="2019-03-27T13:00WEIGHTS"
        >
          <div
            className="app_grid_day_action"
          >
            <img
              alt="WEIGHTS"
              className="app_grid_day_action_icon"
              src="IMAGE_MOCK"
              srcSet="IMAGE_MOCK 2x"
            />
          </div>
          <span
            className="app_grid_day_time"
          >
            1:00PM
          </span>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="app_main_day"
    key="2019-03-29"
  >
    <h2
      className="app_main_day_date"
    >
      March 29
    </h2>
    <h2
      className="app_main_day_weekday"
    >
      Friday
    </h2>
    <div
      className="app_main_day_roll_wrap"
    >
      <div
        className="app_main_day_roll"
      >
        <div
          className="app_main_day_activity"
          key="2019-03-29T11:00SURFING"
        >
          <div
            className="app_grid_day_action"
          >
            <img
              alt="SURFING"
              className="app_grid_day_action_icon"
              src="IMAGE_MOCK"
              srcSet="IMAGE_MOCK 2x"
            />
          </div>
          <span
            className="app_grid_day_time"
          >
            11:00AM
          </span>
        </div>
      </div>
    </div>
  </section>,
]
`;
